### PR TITLE
Register and document volumes.kubernetes.io/controller-managed-attach-detach annotation

### LINKED ISSUE #36656: Annotation `volumes.kubernetes.io/controller-managed-attach-detach` not registered / documented
**This is a Bug Report**

**Problem:**
I have a node in Minikube that's annotated with the key `volumes.kubernetes.io/controller-managed-attach-detach`.

This looks intended, but https://kubernetes.io/docs/reference/labels-annotations-taints/ doesn't document that annotation.

**Proposed Solution:**
Document that annotation; as a side effect, this registers it.

https://docs.okd.io/3.11/install_config/persistent_storage/enabling_controller_attach_detach.html has some tips on what the annotation means.

**Page to Update:**
https://kubernetes.io/docs/reference/labels-annotations-taints/

**Additional Information:**
/sig storage
/language en
/priority backlog

PS. Maybe, as a separate detail, the project should deprecate this annotation. I'm not sure if modern Kubernetes still needs it.

### PR DESCRIPTION
**Pull request by Kartik494**
Fixes #36656 

This PR registered `volumes.kubernetes.io/controller-managed-attach-detach` annotation.

### CODE REVIEW

**Comment by Kartik494** [comment]
cc @sftim @jimangel Please take a look 
Thanks!!

**Comment by sftim** [comment]
/retitle Register and document volumes.kubernetes.io/controller-managed-attach-detach annotation

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -355,6 +355,12 @@ Used on: PersistentVolumeClaim
 
 This annotation will be added to dynamic provisioning required PVC.
 
+### volumes.kubernetes.io/controller-managed-attach-detach
+
+Used on: Node
+
+If a node has set the annotation volumes.kubernetes.io/controller-managed-attach-detach on itself, then its attach and detach operations are being managed by the controller. The controller will automatically inspect all nodes for this annotation and act according to whether it is present or not.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:362]: I'd write:
```markdown
If a node has set the annotation `volumes.kubernetes.io/controller-managed-attach-detach`
on itself, then its storage attach and detach operations are being managed
by the FIXME
{{< glossary_tooltip text="controller" term_id="controller" >}}.
The FIXME controller automatically inspects all nodes for this annotation
and adapts its behavior according to whether the annotation is present or not.

The value of the annotation isn't important; if this annotation exists on a node,
then attach and detached are controller managed.
```

We should explain _which_ controller is doing that volume attach / detach management.

**Review by sftim — COMMENTED** [review]
Thanks.

I recommend some tweaks before we merge this.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 1 above)
Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: @sftim  Could you please take a look here.

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: OK to mark this as resolved.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -355,6 +355,19 @@ Used on: PersistentVolumeClaim
...
 This annotation will be added to dynamic provisioning required PVC.
 
+### volumes.kubernetes.io/controller-managed-attach-detach
+
+Used on: Node
+
+If a node has set the annotation `volumes.kubernetes.io/controller-managed-attach-detach`
+on itself, then its storage attach and detach operations are being managed
+by the controller running on the cluster’s master.
+The cluster’s master controller automatically inspects all nodes for this annotation
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:365]: > The cluster’s master controller automatically inspects all nodes for this annotation

It's not the “master controller”. @Kartik494 would you be willing to ask SIG Storage folks which controller it actually is?

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: Some candidates: https://github.com/kubernetes/kubernetes/tree/master/pkg/controller/volume

Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: Sure i will do this

> > The cluster’s master controller automatically inspects all nodes for this annotation
> 
> It's not the “master controller”. @Kartik494 would you be willing to ask SIG Storage folks which controller it actually is?

Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: @kubernetes/sig-storage-pr-reviews i am a little confuse on opting for which controller inspect all nodes for this annotation. Could you please help me out.

Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: @sftim can you ping sig-storage pr reviewers to take a look into this,

Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: @Kartik494 would **_you_** be willing to ask SIG Storage folks which controller it actually is?

Comment (jsafrane) [reply content/en/docs/reference/labels-annotations-taints/_index.md:365]: It's attach/detach controller running in kube-controller-manager who inspects and acts on the annotation, for example [here](https://github.com/kubernetes/kubernetes/blob/6bdda2da160043c5fcdfd25bbea9e1c1b804c9e5/pkg/controller/volume/attachdetach/attach_detach_controller.go#L899)

And node annotates itself based on `kubelet --enable-controller-attach-detach` cmdline flag or via corresponding kubelet config file field.

Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: @jsafrane thanks for suggestion on this , @sftim could you please take a look and let me know if this sounds good to you.

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -355,6 +355,17 @@ Used on: PersistentVolumeClaim
...
+### volumes.kubernetes.io/controller-managed-attach-detach
+
+Used on: Node
+
+If a node has set the annotation `volumes.kubernetes.io/controller-managed-attach-detach`
+on itself, then its storage attach and detach operations are being managed
+by the attach/detach controller running in kube-controller-manager.
+
+The value of the annotation isn't important; if this annotation exists on a node,
+then attach and detached are controller managed.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:367]:
```suggestion
then storage attaches and detaches are controller managed.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -355,6 +355,17 @@ Used on: PersistentVolumeClaim
 
 This annotation will be added to dynamic provisioning required PVC.
 
+### volumes.kubernetes.io/controller-managed-attach-detach
+
+Used on: Node
+
+If a node has set the annotation `volumes.kubernetes.io/controller-managed-attach-detach`
+on itself, then its storage attach and detach operations are being managed
+by the attach/detach controller running in kube-controller-manager.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:364]:
```suggestion
by the _volume attach/detach_
{{< glossary_tooltip text="controller" term_id="controller" >}} running within the
{{< glossary_tooltip term_id="kube-controller-manager" text="kube-controller-manager" >}}.
```

Some people call it the “attach/detach controller”; I prefer to include “volume” for clarity.

**Review by sftim — COMMENTED** [review]
/lgtm

Making this change means that the docs are less wrong than they are right now.

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Thanks @jsafrane

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 3 above)
Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Done

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 4 above)
Comment (Kartik494) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: Addressed the changes

**Comment by Kartik494** [comment]
@sftim PTAL!

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 5]
```diff
@@ -355,6 +355,19 @@ Used on: PersistentVolumeClaim
 
 This annotation will be added to dynamic provisioning required PVC.
 
+### volumes.kubernetes.io/controller-managed-attach-detach
+
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:359]: It'd be nice to have an example. That can come in a follow-up PR.

**Review by sftim — COMMENTED** [review]
/approve

**Pull request merged**